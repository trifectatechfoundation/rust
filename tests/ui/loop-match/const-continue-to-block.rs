#![allow(incomplete_features)]
#![feature(loop_match)]
#![crate_type = "lib"]

fn const_continue_to_block() -> u8 {
    let state = 0;
    #[loop_match]
    loop {
        state = 'blk: {
            match state {
                0 => {
                    #[const_continue]
                    break 'blk 1;
                }
                _ => 'b: {
                    #[const_continue]
                    break 'b 2;
                    //~^ ERROR `#[const_continue]` must break to a labeled block that participates in a `#[loop_match]`
                }
            }
        }
    }
}
