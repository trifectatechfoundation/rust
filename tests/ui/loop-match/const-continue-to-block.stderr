error: `#[const_continue]` must break to a labeled block that participates in a `#[loop_match]`
  --> $DIR/const-continue-to-block.rs:17:27
   |
LL |                     break 'b 2;
   |                           ^^

error: aborting due to 1 previous error

